_posts/2022-06-08-#22778.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: pr
+date: 2022-06-08
+title: "net processing: Reduce resource usage for inbound block-relay-only connections"
+pr: 22778
+authors: [jnewbery]
+components: ["p2p"]
+host: larryruane
+status: upcoming
+commit: 9db82f1
+---
+
+## Notes
+
+- Each of our node's peers is either inbound (our
+  peer initiated the connection) or outbound (we initiated the connection).
+  Once the connection is established, the relationship is symmetric (the
+  protocol works the same in both directions).
+
+- When we initiate an outbound connection, we may decide that we won't
+  relay (forward) transactions or addresses to this peer.
+  This peer is designated a `block-relay-only` peer.
+  This kind of peer connection was covered in earlier review club meetings
+  (PRs [15759](https://bitcoincore.reviews/15759),
+  [19858](https://bitcoincore.reviews/19858),
+  [20726](https://bitcoincore.reviews/20726)).
+  The peer
+  doesn't know that we're treating it as `block-relay-only`.
+
+- When connecting with a peer, we allocate a
+  [`Peer`](https://github.com/bitcoin/bitcoin/blob/1b2e1d179c5b350cac69ee670e01355314f25e11/src/net_processing.cpp#L203)
+  object.
+  If we plan to announce and send transactions and addresses to
+  this peer (for example, if it's not a `block-relay-only` peer),
+  we create an additional
+  [`TxRelay`](https://github.com/bitcoin/bitcoin/blob/1b2e1d179c5b350cac69ee670e01355314f25e11/src/net_processing.cpp#L250)
+  object (linked from the `Peer` object) to track
```

Comment:
Perhaps 
```suggestion
  [`TxRelay`](https://github.com/bitcoin/bitcoin/blob/1b2e1d179c5b350cac69ee670e01355314f25e11/src/net_processing.cpp#L250)
  struct (owned by the `Peer` object) to track
```

since TxRelay is a data structure rather than a class, and the relationship is that Peer owns the TxRelay struct.